### PR TITLE
[M21] Idol Of Endurance unfinished work (not sure how to proceed)

### PR DESCRIPTION
**Pull request by htrajan**
Not sure how to get this over the finish line. If anything here is useful, feel free to work off of this branch. Otherwise, feel free to delete this PR and do the card from scratch.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/i/IdolOfEndurance.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+    @Override
+    public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
+        ExileZone exileZone = CardUtil.getExileIfPossible(game, exileZoneSource);
+        Card card = game.getCard(objectId);
+        IdolOfEnduranceWatcher watcher = game.getState().getWatcher(IdolOfEnduranceWatcher.class, source.getId());
+        if (card != null && watcher != null) {
+            if (!watcher.usedIdolOfEnduranceExileCast() && exileZone.getCards(StaticFilters.FILTER_CARD_CREATURE, game).contains(card)) {
+                allowCardToPlayWithoutMana(objectId, source, affectedControllerId, game);
+                watcher.setUsedIdolOfEnduranceExileCast(true);
```
Comment (htrajan) [inline Mage.Sets/src/mage/cards/i/IdolOfEndurance.java:121]: this probably doesn't make any sense

File: Mage.Sets/src/mage/cards/i/IdolOfEndurance.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null) {
+            controller.getGraveyard().stream()
+                .map(game::getCard)
+                .filter(Objects::nonNull)
+                .filter(card -> filter.match(card, game))
+                .forEach(card -> controller.moveCards(card, Zone.EXILED, source, game));
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/i/IdolOfEndurance.java:91]: the cards should all be moved at once, so create a `Cards` object, add everything to it and then move that whole collection at once.

Comment (htrajan) [reply Mage.Sets/src/mage/cards/i/IdolOfEndurance.java:91]: noted for the future, thanks

File: Mage.Sets/src/mage/cards/i/IdolOfEndurance.java (hunk 1 above)
Comment (jeffwadsworth) [reply Mage.Sets/src/mage/cards/i/IdolOfEndurance.java:121]: exileZone needs a null check.

Comment (htrajan) [reply Mage.Sets/src/mage/cards/i/IdolOfEndurance.java:121]: It's actually always non-null

**Pull request closed without merging**